src/convertToObject.js

```diff
@@ -5,8 +5,21 @@
  *
  * @return {object}
  */
+
 function convertToObject(sourceString) {
-  // write your code here
+  const converted = sourceString
+    .split(';')
+    .map((el) => el.trim())
+    .filter((el) => el.length > 0)
+    .reduce((obj, curr) => {
+      const arrValue = curr.split(':').map((el) => el.trim());
+
+      obj[arrValue[0]] = arrValue[1];
```

Comment:
Ensure that `arrValue[0]` and `arrValue[1]` exist before assigning them to `obj`. If the input string is not formatted correctly, this could lead to undefined values being assigned as keys or values in the object.